src/wallet/message_store.cpp

```diff
@@ -661,31 +712,36 @@ void message_store::delete_all_messages()
   m_messages.clear();
 }
 
-// Make a message text, which is "attacker controlled data", reasonably safe to display
+// Make a text, which is "attacker controlled data", reasonably safe to display
 // This is mostly geared towards the safe display of notes sent by "mms note" with a "mms show" command
-void message_store::get_sanitized_message_text(const message &m, std::string &sanitized_text) const
+void message_store::get_sanitized_text(const std::string &text, size_t max_length, std::string &sanitized_text) const
 {
-  sanitized_text.clear();
-
   // Restrict the size to fend of DOS-style attacks with heaps of data
-  size_t length = std::min(m.content.length(), (size_t)1000);
+  size_t length = std::min(text.length(), max_length);
+  sanitized_text = text.substr(0, length);
```

Comment:
I am not sure what you mean with "returning by value". Would that be only a modification of those two lines somehow, or also involve the call to `utf8canonical` on the next line?

Reply:
Return the `std::string` instead of an out parameter (`sd::string&`). There's no benefit to the current implementation, the memory for `sanitized_text` is never being re-used. The `substr` call is replacing the internal buffer and so is the call to `utf8canonical`.

 All of the calls to the function become "natural" assignments - move assignments or construct-in-place operations (return-value-optimization). Currently a number of calls to this function copy the string 3 times (`substr`, `utf8canonical`, and a copy assignment from temp `std::string`). The `max_length` keeps the copying to a minimum, at least.

Reply:
Done.